File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/WindowElements/addwindowelement.rb
module Asposediagramjava
  module AddWindowElement
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # initialize window object
        window = Rjb::import('com.aspose.diagram.Window').new
        
        # set window state
        window.setWindowState(Rjb::import('com.aspose.diagram.WindowStateValue').MAXIMIZED)
        
        # set window height
        window.setWindowHeight(500)
        
        # set window width
        window.setWindowWidth(500)
        
        # set window type
        window.setWindowType(Rjb::import('com.aspose.diagram.WindowTypeValue').STENCIL)
        
        # add window object
        diagram.getWindows().add(window)

        # save in any supported format
        diagram.save(data_dir + "AddWindowElement.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Added window element to the visio instance."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithWindowElements/AddWindowElement.php
<?php
namespace Aspose\Diagram\WorkingwithWindowElements;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\Window as Window;
use com\aspose\diagram\WindowStateValue as WindowStateValue;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
use com\aspose\diagram\WindowTypeValue as WindowTypeValue;

class AddWindowElement{
    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vsd");

        # initialize window object
        $window = new Window();

        # set window state
        $windowStateValue=new WindowStateValue();
        $window->setWindowState($windowStateValue->MAXIMIZED);

        # set window height
        $window->setWindowHeight(500);

        # set window width
        $window->setWindowWidth(500);

        # set window type
        $windowTypeValue=new WindowTypeValue();
        $window->setWindowType($windowTypeValue->STENCIL);

        # add window object
        $diagram->getWindows()->add($window);

        # save in any supported format
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."AddWindowElement.vdx", $saveFileFormat->VDX);

        print "Added window element to the visio instance.".PHP_EOL;
    }
}